dht.go

```diff
@@ -400,3 +402,18 @@ func (dht *IpfsDHT) protocolStrs() []string {
 func mkDsKey(s string) ds.Key {
 	return ds.NewKey(base32.RawStdEncoding.EncodeToString([]byte(s)))
 }
+
+func (dht *IpfsDHT) PeerId() peer.ID {
+	return dht.self
+}
+
+func (dht *IpfsDHT) PeerKey() []byte {
+	return kb.ConvertPeerID(dht.self)
+}
+
+func (dht *IpfsDHT) SetClientMode() {
```

Comment:
Let's add a unit test for this.

Reply:
Actually this API is asymmetric as it allows to demote a server to a client, but not promote a client to a server. Let's change this to `SetMode()` with an enum value, so it's more useful?

Reply:
A test is good, perhaps in a later PR. I'm not sure about getting fancy with flags and such, there's no way we can know that there's only 2 modes. Currently I've just exposed what I need, which will act correctly for any user without confusion. Whether or not other modes are added later, this method will be unaffected. See https://www.martinfowler.com/bliki/FlagArgument.html and https://softwareengineering.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior as examples.

Reply:
I'm not convinced. Note that anything that you add now forms part of the public API and downstreams will start using it. If you're knowingly adding an ad-hoc method here for a short-term need, you're essentially piling up technical debt from the get-go.

My point: you can expose what you need in a way that paves our _near_ future needs. In fact, I not only **easily** see a third _bootstrap_ mode, but in fact I strongly suggest adding it soon. In this mode, the node would only service `FindPeer` requests, ignoring calls to store and get data and provider records. We need that, and I do not like adding N `SetXYZMode()` methods because the semantics of that is even more confusing (nothing implies they are exclusive of one another).

Reply:
> A test is good, perhaps in a later PR.

Why a later PR? Just add it here as a commit.